coordinator/persistence-api/src/main/java/io/stargate/db/metrics/api/ClientInfoMetricsTagProvider.java

```diff
@@ -28,9 +32,35 @@
  */
 public interface ClientInfoMetricsTagProvider {
 
-  /** Returns default interface implementation, which returns empty tags for all methods. */
+  /**
+   * Returns default interface implementation, which returns empty tags or with driver info for all
+   * methods.
+   */
   ClientInfoMetricsTagProvider DEFAULT = new ClientInfoMetricsTagProvider() {};
 
+  /**
+   * Returns tags for a {@link ClientInfo}.
+   *
+   * <p>Note that the implementation must return constant amount of tags for any input. IF the
+   * client info is populated with driver info, the tags should be populated with the driver name
+   * and version. Otherwise, the tags should be populated with "unknown" values.
+   *
+   * @param clientInfo {@link ClientInfo}
+   * @return Tags
+   */
+  default Tags getClientInfoTagsByDriver(ClientInfo clientInfo) {
+    List<Tag> tags = new ArrayList<>(2);
+    if (clientInfo.driverInfo().isPresent()) {
+      DriverInfo driverInfo = clientInfo.driverInfo().get();
+      tags.add(Tag.of("driverName", driverInfo.name()));
+      tags.add(Tag.of("driverVersion", driverInfo.version().orElse("unknown")));
+    } else {
+      tags.add(Tag.of("driverName", "unknown"));
+      tags.add(Tag.of("driverVersion", "unknown"));
```

Comment:
* for the unknown value we should use the constant `StargateMetricConstants.UNKNOWN`
* `driverName` and `driverVersion` could be the public constants of this class, so implementing classes can override those if needed